app/src/components/market/sections/market_list/market_home_container.tsx

```diff
@@ -106,32 +107,43 @@ const MarketHomeContainer: React.FC = () => {
   useEffect(() => {
     if (loading) {
       setMarkets(markets => (RemoteData.hasData(markets) ? RemoteData.reloading(markets.data) : RemoteData.loading()))
-    } else if (error) {
-      setMarkets(RemoteData.failure(error))
     } else if (fetchedMarkets) {
       const { fixedProductMarketMakers } = fetchedMarkets
 
       setMarkets(RemoteData.success(wrangleResponse(fixedProductMarketMakers, context.networkId)))
-      if (fixedProductMarketMakers.length === 0) {
+
+      if (fixedProductMarketMakers.length < PAGE_SIZE) {
         setMoreMarkets(false)
       }
+
+      setIsFiltering(false)
+    } else if (error) {
+      setMarkets(RemoteData.failure(error))
+      setIsFiltering(false)
     }
-  }, [fetchedMarkets, loading, error, context.networkId])
+  }, [fetchedMarkets, loading, error, context.networkId, markets])
 
   const onFilterChange = useCallback((filter: any) => {
-    setMoreMarkets(true)
```

Comment:
Why is this removed?

Reply:
It was showing the button when it was not needed IIRC.

Generally speaking, my intention was to not to display the button unless there are more than 10 items. The way it was implemented before it was always displayed, and hidden when trying to load more elements and finding there were not more than 10, or no more items to load.